src/partisan/irods.py

```diff
@@ -1976,14 +1978,25 @@ def iter_contents(
         )
 
         contents = [_make_rods_item(item, pool=self.pool) for item in items]
-        for elt in contents:
-            elt.path = self.path / elt.path  # Make an absolute path
-            yield elt
+        contents.sort()
 
-            if isinstance(elt, Collection):
-                yield from elt.iter_contents(
-                    acl=acl, avu=avu, timeout=timeout, tries=tries
-                )
+        if recurse:
+            for elt in contents:
+                elt.path = self.path / elt.path  # Make an absolute path
+                yield elt
+
+                if isinstance(elt, Collection):
+                    yield from elt.iter_contents(
+                        acl=acl,
+                        avu=avu,
+                        recurse=recurse,
+                        timeout=timeout,
+                        tries=tries,
+                    )
+        else:
+            for elt in contents:
+                elt.path = self.path / elt.path
+                yield elt
```

Comment:
```suggestion

        for elt in contents:
            elt.path = self.path / elt.path  # Make an absolute path
            yield elt
            if recurse:
                if isinstance(elt, Collection):
                    yield from elt.iter_contents(
                        acl=acl,
                        avu=avu,
                        recurse=recurse,
                        timeout=timeout,
                        tries=tries,
                    )
```
I also suggest that the conditional be inside the for loop, to remove repeated code.

Reply:
Oh, yes. You're right - that was actually a bug, then! Originally I intended it to be non-recursive, but I think I ended up putting some experimental recursive code in there earlier than I wanted to.